### ISSUE TITLE
rustup.sh installation instructions not working

### ISSUE THREAD

**Issue by cardoe** [rust-lang-nursery/rustup.rs#549]
The rustup.sh website says the steps to install are:

```
curl https://sh.rustup.rs -sSf | sh
```

However the SSL certificate that the site uses will cause curl to error with:

curl: (51) SSL: certificate subject name (*.rust-lang.org) does not match target host name 'sh.rustup.rs'.

The site is using the '*.rust-lang.org' certificate when it needs to have its own certificate for sh.rustup.rs or that host needs to be added to the rust-lang.org catch all certificate.

**Comment by brson** [comment]
Thanks for the report. That looks bad!

I can't reproduce this today. @alexcrichton maybe this was temporarily misconfigured while you were working with Let's Encrypt recently?

**Comment by alexcrichton** [comment]
Hm nah unfortunately that was just yesterday and this was a few days before that. @cardoe what version of curl were you using? I believe that because we're using nginx with multiple domains, each of which has a different SSL cert, that we'd only work with clients that support SNI. Some older versions of curl don't support that I believe, and nginx may be defaulting to handing one certificate for clients that don't support SNI.

**Comment by brson** [comment]
I recall old versions on OS X have problems with sh.rustup.rs, but I don't think it's because of SNI.

**Comment by cardoe** [comment]
I'm using curl 7.47.0. I'm on Ubuntu 16.04. I still see the issue right now.

**Comment by cardoe** [comment]
Its built against GnuTLS 3.4.10 instead of OpenSSL.

**Comment by alexcrichton** [comment]
Whoa that should be plenty new enough, could you try getting more verbose output out of curl?

**Comment by cardoe** [comment]
Well good news (and bad news for me). The certificate I'm seeing when I get the failure is an EssentialSSL certificate. Which I believe is a Comodo certificate. You guys mentioned Let's Encrypt and I went and checked the certificate from a totally different network and sure enough sh.rustup.rs is serving up a Let's Encrypt certificate. So it sounds like I've got a MITM attack going on. Sorry for making you guys think there was a problem on your end.
